### PR TITLE
fix/intermediate-list-fixing-duplicate-pay-btn

### PR DESCRIPTION
**Pull request by jeppekroghitk**
#### Link to issue

Bugfix for https://platform.dandigbib.org/issues/5391

#### Description

Modifies the conditional rendering of payment button in the intermediate list, to fix a bug causing the button to be displayed twice under certain circumstances.

#### Screenshot of the result

Before:
Screenshot 2023-01-30 at 14 36 43

After:
Screenshot 2023-01-30 at 14 36 19

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

**Review by sinejespersen — APPROVED** [review]

File: src/apps/intermediate-list/stackable-fees/total-payment-pay.tsx [hunk 1]
```diff
@@ -67,15 +68,19 @@ const TotalPaymentPay: FC<TotalPaymentPayProps> = ({
...
+            }
+            disabled={!showPaymentButton}
+          >
+            {t("payText")}
+          </button>
+        )}
+        {postPaymentChangeChecked && (
           <button
             type="button"
             className="btn-primary btn-filled btn-small arrow__hover--right-small mt-16"
```
Comment (spaceo) [inline src/apps/intermediate-list/stackable-fees/total-payment-pay.tsx:86]: I know that this is not related to this change, which is approved anyway.
But I noticed that you use same "base" classes in the two buttons.
So above line 31 you could create a variable like this:
```typescript
const paymentButtonBaseClass = clsx(
    ["btn-primary", "btn-small", "arrow__hover--right-small", "mt-16"],
);
const paymentButtonClass = clsx(
    paymentButtonBaseClass,
    { "btn-outline": !showPaymentButton, "btn-filled": showPaymentButton }
  );
```
and then use the`paymentButtonBaseClass` again in:
```typescript
{postPaymentChangeChecked && (
          <button
            type="button"
            className={paymentButtonBaseClass}

**Review by spaceo — APPROVED** [review]
Approved. But please consider comment 👍
